internal/resources/machinelearning/resource_alert.go

```diff
@@ -277,6 +277,13 @@ func (r *alertResource) Update(ctx context.Context, req resource.UpdateRequest,
 	if resp.Diagnostics.HasError() {
 		return
 	}
+	// It is required to fetch the ID from the state data. It is not present in the plan data.
```

Comment:
You can use the `UseStateForUnknown` modifier on the id attribute instead: https://github.com/grafana/terraform-provider-grafana-adaptive-metrics/blob/main/internal/provider/segment_resource.go#L59-L61

Reply:
Ohh thanks! I have updated the PR to use this.